cnd/src/swap_protocols/facade.rs

```diff
@@ -149,63 +151,108 @@ impl Facade {
+                    herc20_asset: swap.alpha,
+                    halight_asset: swap.beta,
+                });
+            }
         };
 
-        let herc20_params = Load::<herc20::InProgressSwap>::load(self, id).await?;
-        let halight_params = Load::<halight::InProgressSwap>::load(self, id).await?;
+        let (
+            herc20_asset,
+            herc20::Identities {
+                redeem_identity: herc20_redeem_identity,
+                refund_identity: herc20_refund_identity,
+            },
+            herc20_expiry,
+        ) = self.load(id).await?;
+        let (
+            halight_asset,
+            halight::Identities {
+                redeem_identity: halight_redeem_identity,
+                refund_identity: halight_refund_identity,
+            },
+            cltv_expiry,
+        ) = self.load(id).await?;
 
         let secret = self.seed.derive_swap_seed(id).derive_secret();
 
-        Ok(Some(http_api::AliceHerc20HalightBitcoinSwap {
-            alpha_ledger_state: alpha_state,
-            beta_ledger_state: beta_state,
-            herc20_params,
-            halight_params,
+        Ok(http_api::AliceHerc20HalightBitcoinSwap::Finalized {
+            herc20_asset,
+            herc20_refund_identity,
+            herc20_redeem_identity,
+            herc20_expiry,
+            herc20_state,
+            halight_asset,
+            halight_redeem_identity,
+            halight_refund_identity,
+            cltv_expiry,
+            halight_state,
             secret,
-        }))
+        })
     }
 
     pub async fn get_bob_herc20_halight_swap(
         &self,
         id: LocalSwapId,
-    ) -> anyhow::Result<Option<http_api::BobHerc20HalightBitcoinSwap>> {
+    ) -> anyhow::Result<http_api::BobHerc20HalightBitcoinSwap> {
         let alpha_state = self.herc20_states.get(&id).await?;
         let beta_state = self.halight_states.get(&id).await?;
 
-        let (alpha_state, beta_state) = match (alpha_state, beta_state) {
+        let (herc20_state, halight_state) = match (alpha_state, beta_state) {
             (Some(alpha_state), Some(beta_state)) => (alpha_state, beta_state),
-            _ => return Ok(None),
+            _ => {
+                let swap: Swap<asset::Erc20, asset::Bitcoin> = self.db.load(id).await?;
+                return Ok(http_api::BobHerc20HalightBitcoinSwap::Created {
+                    herc20_asset: swap.alpha,
+                    halight_asset: swap.beta,
+                });
+            }
         };
 
-        let herc20_params = Load::<herc20::InProgressSwap>::load(self, id).await?;
-        let halight_params = Load::<halight::InProgressSwap>::load(self, id).await?;
+        let (
+            herc20_asset,
+            herc20::Identities {
+                redeem_identity: herc20_redeem_identity,
+                refund_identity: herc20_refund_identity,
+            },
+            herc20_expiry,
```

Comment:
The tuple idea is quite nice but there is a downside to using tuples instead of types; now every dev has to look up the order of the three types in the tuple each time they write any code like this.

Reply:
I totally agree and we considered a few other possibilities. For now I am happy with this, especially since we avoid the ambiguity between the two identities by introducing the `{herc20,halight}::Identities` types.